### ISSUE TITLE
Discord announcement fails

### ISSUE THREAD

**Issue by crschnick** [jreleaser/jreleaser#1347]
The discord announcement failed for me while for example the slack one doesn't. There is no stack trace, only a warning:   

    [INFO]  JReleaser 1.5.1
    [INFO]    - basedir set to /home/runner/work/xpipex/xpipex/xpipe/dist
    [INFO]    - outputdir set to /home/runner/work/xpipex/xpipex/xpipe/dist/build/jreleaser
    [INFO]  git-root-search set to true
    [WARN]  Auto configure detected 'xpipex' as repository name but project has 'xpipe' configured
    [INFO]  Validating configuration
    [INFO]  Strict mode set to false
    [INFO]  Project version set to 0.5.36
    [INFO]  Release is not snapshot
    [INFO]  Timestamp is 2023-04-21T07:33:08.14937065Z
    [INFO]  HEAD is at 47c1f4a
    [INFO]  Platform is linux-x86_64
    [INFO]  dry-run set to false
    [INFO]  Generating changelog
    [INFO]  Storing changelog: build/jreleaser/release/CHANGELOG.md
    [INFO]  Cataloging artifacts
    [INFO]    [sbom] Cataloging is not enabled. Skipping
    [INFO]  Calculating checksums for distributions and files
    [INFO]    [checksum] All checksums are up-to-date. Skipping
    [INFO]  Signing distributions and files
    [INFO]    [sign] Signing is not enabled. Skipping
    [INFO]  Deploying Maven artifacts
    [INFO]    [maven] Deploying is not enabled. Skipping
    [INFO]  Uploading distributions and files
    [INFO]    [upload] Uploading is not enabled. Skipping
    [INFO]  Releasing to https://github.com/xpipe-io/xpipe
    [INFO]   - uploading changelog.md
    [INFO]   - uploading checksums_sha256.txt
    [INFO]   - uploading xpipe-api-0.5.36.jar
    [INFO]   - uploading xpipe-beacon-0.5.36.jar
    [INFO]   - uploading xpipe-core-0.5.36.jar
    [INFO]   - uploading xpipe-installer-linux-x86_64.deb
    [INFO]   - uploading xpipe-installer-linux-x86_64.rpm
    [INFO]   - uploading xpipe-installer-macos-x86_64.pkg
    [INFO]   - uploading xpipe-installer-windows-x86_64.msi
    [INFO]   - uploading xpipe-portable-linux-x86_64.tar.gz
    [INFO]   - uploading xpipe-portable-macos-x86_64.dmg
    [INFO]   - uploading xpipe-portable-windows-x86_64.zip
    [INFO]  Writing output properties to build/jreleaser/output.properties
    [INFO]  JReleaser succeeded after 29.376 s
    
    > Task :dist:jreleaserAnnounce
    [INFO]  JReleaser 1.5.1
    [INFO]    - basedir set to /home/runner/work/xpipex/xpipex/xpipe/dist
    [INFO]    - outputdir set to /home/runner/work/xpipex/xpipex/xpipe/dist/build/jreleaser
    [INFO]  git-root-search set to true
    [WARN]  Auto configure detected 'xpipex' as repository name but project has 'xpipe' configured
    [INFO]  Validating configuration
    [INFO]  Strict mode set to false
    [INFO]  Project version set to 0.5.36
    [INFO]  Release is not snapshot
    [INFO]  Timestamp is 2023-04-21T07:33:08.14937065Z
    [INFO]  HEAD is at 47c1f4a
    [INFO]  Platform is linux-x86_64
    [INFO]  dry-run set to false
    [INFO]  Generating changelog
    [INFO]  Storing changelog: build/jreleaser/release/CHANGELOG.md
    [INFO]  Announcing release
    [INFO]    [slack] channel: announcements
    [INFO]    [webhook.discord] 🚀 X-Pipe 0.5.36 has been released.
    
    ## Changes in 0.5.36
    - Add support to lock your workspace with a custom password similar to password managers.
      This will encrypt all stored sensitive information using the password as the key.
      The general password encryption has also been reworked, which results in all existing passwords becoming invalid.
      So make sure to reenter your passwords for any connection.
    - Add experimental support for PowerShell Remote connections
    - Add support for PowerShell core shell environments
    - Improve file browser performance
    - Improve initial connection startup after being added
    - Implement initial support for future package manager builds
    - Rework updating system to accommodate package managers
    - Improve error handling
    - Improve error display in connection creation
    - Fix many bugs
    
    So don't forget to reenter your passwords for all connection.
    
    
    If you have already installed X-Pipe and have updates enabled, this update will eventually be automatically installed.
    
    You can get installers and portable versions here: https://github.com/xpipe-io/xpipe/releases/tag/0.5.36
    [WARN]    [webhook.discord] Webhook replied with: 400 reason: Bad Request,{"code": 50109, "message": "The request body contains invalid JSON."}
    [INFO]  Writing output properties to build/jreleaser/output.properties
    [INFO]  JReleaser succeeded after 0.389 s

My configuration (the tokens are valid):

        announce {
            webhooks {
                discord {
                    active = 'RELEASE'
                    webhook = proj.hasProperty("XPIPE_DISCORD_WEBHOOK") ? proj.property("XPIPE_DISCORD_WEBHOOK") : System.getenv(
                            "XPIPE_DISCORD_WEBHOOK")
                    messageTemplate = 'misc/discord.tpl'
                }
            }
        }

        announce {
            slack {
                active = 'ALWAYS'
                channel = 'announcements'
                // webhook = proj.hasProperty("XPIPE_SLACK_WEBHOOK") ? proj.property("XPIPE_SLACK_WEBHOOK") : System.getenv("XPIPE_SLACK_WEBHOOK")
                token = proj.hasProperty("XPIPE_SLACK_TOKEN") ? proj.property("XPIPE_SLACK_TOKEN") : System.getenv("XPIPE_SLACK_TOKEN")
                messageTemplate = 'misc/discord.tpl'
            }
        }

**Comment by aalmiray** [comment]
Must be a formatting issue in the announcer's implementation. Could you show the contents of the `discord.tpl` file?

**Comment by crschnick** [comment]
🚀 {{projectName}} {{projectVersion}} has been released.
    
    {{#rawChangelog}}
    {{{rawChangelog}}}
    {{/rawChangelog}}
    
    If you have already installed {{projectName}} and have updates enabled, this update will eventually be automatically installed.
    
    You can get installers and portable versions here: {{releaseNotesUrl}}

**Comment by aalmiray** [comment]
The `discord` announcer has been deprecated. You may configure an explicit webhook announcer like this

```yml
announce:
  webhooks:
    discord:
      active: ALWAYS
      messageProperty: content
      structuredMessage: true
      messageTemplate: discord.tpl
```

More details at https://jreleaser.org/guide/early-access/reference/announce/webhooks.html#_discord

**Comment by crschnick** [comment]
I am using the webhook already

**Comment by aalmiray** [comment]
You're missing 2 properties: `messageProperty` and `structuredMessage`.

**Comment by crschnick** [comment]
So if I don't use these properties it is not using the webhook? I didn't figure this would be an issue as there was no warning.

**Comment by aalmiray** [comment]
No. It's using the webhook but Discord payloads require special formatting. The webhooks announcer is generic and can handle any incoming webhook. Some services require that the content is wrapped in a JSON paylod. some prefer "content", or "text", or "message" as property, that's why we made it configurable. Some other services take the content as is (unstructured).

This is why we have 2 properties for defining how the payload should be formatted.

**Comment by aalmiray** [comment]
There can't be no warning that a generic webhook targeting Discord is not properly configured. The name of the webhook could be anything so we can't match it by name to apply some rules. This is why the docs offer an example on how to configure webhooks for Discord.

**Comment by crschnick** [comment]
Oh I see now. I changed it and it works now.

The documentation says to use `
      messageProperty: text` but you said to use `
      messageProperty: content`, so I'm still a little bit confused about that. The documentation also only explains the purpose of `structuredMessage` vaguely, so even if I copy pasted the discord announcement I would still have no idea why the property is needed just from looking at the documentation.

**Comment by aalmiray** [comment]
Indeed. The docs require and update. Will get to it.
